core/src/main/scala/io/qbeast/core/model/WriteMode.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.qbeast.core.model
+
+/**
+ * Names of possible write modes
+ */
+
+object WriteMode {
```

Comment:
can we use this object to map between other Mode classes? (for Delta, Spark...)

Reply:
Not sure I'm following. What do you mean exactly?

Reply:
I think that in different parts of the codebase we reference the SaveModes from Spark (QbeastDataSource) and from Delta (DeltaMetadataManager)

Should we add to this WriteMode object the mapping between them?

A method like:

`.fromSparkSaveMode(...)
`
or do you think it is best to keep the `WriteMode` as an enumeration an leave the logic to the external components (as it is right now)?

Reply:
In principle, the only place I see a potential replacement of `SaveMode`->`WriteMode` is in the files I modified in this PR, primarily in the `DeltaMetadataManager/Writer`. While I do see `SaveMode` referenced in other files, I think it is not interchangeable.